src/Analysis/GWASUIApp/GWASJob.jsx

```diff
@@ -0,0 +1,177 @@
+import React, { useEffect } from 'react';
+import {
+  Button, List, Tag, Popconfirm,
+} from 'antd';
+import {
+  CheckCircleOutlined,
+  SyncOutlined,
+  CloseCircleOutlined,
+  QuestionCircleOutlined,
+} from '@ant-design/icons';
+import { useQuery, useMutation } from 'react-query';
+import PropTypes from 'prop-types';
+import { gwasWorkflowPath } from '../../localconf';
+import { headers, fetchAndSetCsrfToken } from '../../configs';
+import { getPresignedUrl } from '../AnalysisJob';
+
+const GWASJob = ({ workflow, refreshWorkflows }) => {
+  useEffect(() => {
+    fetchAndSetCsrfToken().catch((err) => { console.log('error on csrf load - should still be ok', err); });
```

Comment:
why is this needed here? isn't this CSRF token set been taken care of in each index page as part of the app init?

Reply:
I think its at least needed in GWASUIApp. removed from children components